src/app/cmdoptions.cpp

```diff
@@ -332,7 +333,7 @@ QBtCommandLineParameters::QBtCommandLineParameters(const QProcessEnvironment &en
     , webUiPort(WEBUI_PORT_OPTION.value(env, -1))
     , torrentingPort(TORRENTING_PORT_OPTION.value(env, -1))
     , skipDialog(SKIP_DIALOG_OPTION.value(env))
-    , profileDir(PROFILE_OPTION.value(env))
+    , profileDir(Utils::Fs::toAbsolutePath(Path(PROFILE_OPTION.value(env))))
```

Comment:
I believe the appropriate subsystems (Profile in this case) should "decide" of how to treat relative paths.

Reply:
The command line parser currently makes torrent paths absolute (a little further down in that file):
```c
            QFileInfo torrentPath;
            torrentPath.setFile(arg);

            if (torrentPath.exists())
                result.torrentSources += torrentPath.absoluteFilePath();
            else
                result.torrentSources += arg;
```
If the rest of the program requires the paths to be absolute, it makes sense to me to treat relative paths as a convenience for humans and convert them to the internal representation at the first opportunity (when parsing arguments).

Reply:
> If the rest of the program requires the paths to be absolute

No, it's not.
If there was a single application-wide policy, I would agree with this point of view.
But as I mentioned above, each subsystem "has the right" to interpret relative paths at its discretion. For example, relative "save paths" are treated by the BitTorrent subsystem in a special way (as relative to some basic "save path", not to the working directory of the application).
So I would even modify the legacy code that you mentioned above so that it doesn't do any conversions.

Reply:
Well, it looks like I've changed my mind about this.